reflex/components/navigation/link.py

```diff
@@ -42,11 +42,13 @@ def create(cls, *children, href: Optional[Var] = None, **props) -> Component:
             **props: The props of the component.
 
         Raises:
-            ValueError: in case of missing children
+            ValueError: in case of missing children or link
 
         Returns:
             Component: The link component
         """
+        if href and not isinstance(href, str):
+            raise ValueError("Invalid href passed in Link. Expected a 'str' for href.")
         if href and not len(children):
             raise ValueError("Link without a child will not display")
         elif href is None and len(children):
```

Comment:
```suggestion
        elif not href and len(children):
```

Changing the condition here to a falsey check will actually fix the issue seen in #1486, where the user supplies `href=""`